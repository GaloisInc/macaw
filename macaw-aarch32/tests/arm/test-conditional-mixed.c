#include "util.h"

int g = -11;

void __attribute__((noinline, target("thumb"))) f() {
  if(g > 0) {
    g = g + 1;
  }

  g = g - 1090;
}

void _start() {
  // Making this into a call to ensure that the conditional is thumb-ified
  f();
  EXIT();
}
