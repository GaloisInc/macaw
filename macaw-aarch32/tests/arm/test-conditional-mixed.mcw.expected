R { funcs = [ (0x100d9, [ (0x100d9, 12)
                        , (0x100e4, 28)
                        , (0x100ee, 18)
                        ])
            , (0x10108, [ (0x10108, 12)
                        , (0x10114, 8)
                        ])
            , (0x10128, [ (0x10128, 8)])
            ]
  , ignoreBlocks = [0x1011c]
  }
